### ISSUE TITLE
Sensor errors should be sent as error codes, not strings

### ISSUE THREAD

**Issue by gordonbrander** [OpenAgInitiative/openag_brain#76]
Was getting this error:

> [ERROR] [WallTime: 1376891179.350615] Message from device dropped: message larger than buffer.

Per http://wiki.ros.org/rosserial/Overview/Limitations, I thought we had maxed out our 25 pubs/25 subs, but that wasn't the case (only had 15).

It turned out to be sensor errors. All of the errors go through a single topic. The messages on it are just long enough to overflow the buffer sometimes.

We should send error codes (as float) over the wire instead of error strings.

**Comment by gordonbrander** [comment]
From https://github.com/OpenAgInitiative/openag_python/blob/master/docs/firmware_modules.rst:

> The :cpp:class:`Module` superclass defines a :cpp:member:`status_level` attribute which the firmware module should use to report its current status. Valid value for this attribute (all defined in the header file for the :cpp:class:`Module` superclass) are :c:data:`OK` (which means that the module is "ok"), :c:data:`WARN` (which means that there is some warning for the module), and :c:data:`ERROR` (which means that there is an error preventing the module from working as desired). The superclass also defines a :cpp:member:`status_msg` attribute which is a :cpp:class:`String` that the firmware module should use to describe the status of the module. This is generally an empty string when the status level is "ok" and an error message when the status level is "warn" or "error".

**Comment by gordonbrander** [comment]
@LeonChambers proposal: change `status_msg` to `status_code`. By convention, `module.json`, defines a `status_code` object which is used to translate status codes to messages for nice logging:

```json
{
  ...
  "status_code": {
    "2000": "OK",
    "5000": "Sensor responded with the wrong function code",
    "5001": "Sensor did not return enough information"
  }
}
```

Status codes are arbitrary `int` values chosen by the module author to be relevant within the context of the module.

**Comment by LeonChambers** [comment]
Sounds good to me.

For reference, the current convention was chosen for easy integration with the ROS [diagnostics](http://wiki.ros.org/diagnostics) stack, specifically for populating [DiagnosticStatus](http://docs.ros.org/api/diagnostic_msgs/html/msg/DiagnosticStatus.html) messages. But to minimize overhead on the serial connection, it makes sense to adopt this new format and have something on the ROS side translate status codes to messages.

**Comment by gordonbrander** [comment]
Maybe error codes should be `uint8_t` (0..255)

**Comment by gordonbrander** [comment]
Pull requests for an interim fix:

- https://github.com/OpenAgInitiative/openag_atlas_rgb/pull/2
- https://github.com/OpenAgInitiative/openag_dht22/pull/3
- https://github.com/OpenAgInitiative/openag_ds18b20/pull/2
- https://github.com/OpenAgInitiative/openag_am2315/pull/2
- https://github.com/OpenAgInitiative/openag_atlas_do/pull/1
- https://github.com/OpenAgInitiative/openag_atlas_ec/pull/1
- https://github.com/OpenAgInitiative/openag_atlas_orp/pull/1
- https://github.com/OpenAgInitiative/openag_atlas_ph/pull/1
- https://github.com/OpenAgInitiative/openag_gc0012/pull/2
- https://github.com/OpenAgInitiative/openag_mhz19/pull/1
- https://github.com/OpenAgInitiative/openag_bh1750/pull/1
- https://github.com/OpenAgInitiative/openag_pwm_actuator/pull/1
- https://github.com/OpenAgInitiative/openag_mhz16/pull/2

**Comment by LeonChambers** [comment]
Fixed in #102
